src/pages/index.js

```diff
@@ -28,44 +30,146 @@ const enableValidation = (config) => {
 
 enableValidation(config);
 
-//Create Card Section
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: (cardData) => {
-      const cardElement = createNewCard(cardData, "#card-template");
-      cardSection.addItem(cardElement);
-    },
+//Create API
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "0e6b8e99-3fb6-490c-b2a5-5d193b29415c",
+    "Content-Type": "application/json",
   },
-  "elements__list"
-);
+});
 
-cardSection.renderItems();
+//Card Actions
+let cardSection;
+
+api
+  .getInitialCards()
+  .then((cards) => {
+    cardSection = new Section(
+      {
+        items: cards,
+        renderer: (cardData) => {
+          const cardElement = createNewCard(cardData, "#card-template");
+          cardSection.addItem(cardElement);
+        },
+      },
+      "elements__list"
+    );
+    cardSection.renderItems();
+  })
+  .catch((err) => {
+    console.error(err);
+  });
 
 //New Card Functions
 function createNewCard(cardData, template) {
-  return new Card(cardData, template, handleImageClick).generateCard();
+  return new Card(
+    cardData,
+    template,
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick
+  ).generateCard();
+}
+
+function handleLikeClick(cardId, likeStatus) {
+  if (likeStatus) {
+    api
+      .likeCard({ _id: cardId })
+      .then(() => {
+        console.log("This post has been liked");
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  } else {
+    api
+      .dislikeCard({ _id: cardId })
+      .then(() => {
+        console.log("This post has been disliked");
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+}
+
+function handleDeleteClick(cardId, cardElement) {
+  confirmDelete.open(cardId, cardElement);
+}
+
+function handleDeleteSubmit(cardId, cardElement) {
+  cardElement.remove();
```

Comment:
Needs correction: you should remove the card only after the successful api response